### PR TITLE
Updated Installer

### PR DESCRIPTION
**Pull request by Itorius**
The installer now searches for the Terraria install directory, this only works on Windows.

### CODE REVIEW

File: installer2/WindowsInfo.java [hunk 1]
```diff
@@ -1,47 +1,67 @@
...
-            {
-                directory = getInstallDir(path);
+
+        File InputVDFFile = new File(steamPath + "/config/config.vdf");
+        VDF InputVDF = new VDF(InputVDFFile);
+        BufferedReader buffer = new BufferedReader(new FileReader(steamPath + "/config/config.vdf"));
+
+        String line = null;
+        while ((line = buffer.readLine()) != null) {
+            Pattern pattern = Pattern.compile("\"(.*?)\"");
```
Comment (Chicken-Bones) [inline installer2/WindowsInfo.java]: could be outside the loop, but not actually likely to be a performance issue in this case

File: installer2/WindowsInfo.java [hunk 2]
```diff
@@ -1,47 +1,67 @@
...
+            Matcher matcher = pattern.matcher(line);
+
+            while (matcher.find()) {
+                String key = matcher.group(1);
+                if (key.contains("BaseInstallFolder")) {
```
Comment (Chicken-Bones) [inline installer2/WindowsInfo.java]: in general I'd recommend inverting conditions and using continue to reduce nesting, but it's fine here

**Comment by Chicken-Bones** [comment]
Seems fine to me, and you used a package. I'd just like someone else to test it

**Comment by JavidPack** [comment]
Why did you close this @Itorius ?

**Comment by Itorius** [comment]
Woops, I accidentally removed the commits from my repo

**Comment by JavidPack** [comment]
createInstallers.bat needs to be fixed for this to be accepted. I don't know how to build the installer.

**Pull request closed without merging**